pkg/services/sqlstore/migrations/accesscontrol/permission_migrator.go

```diff
@@ -210,19 +207,10 @@ func batch(count, batchSize int, eachFn func(start, end int) error) error {
 	return nil
 }
 
-func generateNewRoleUID(sess *xorm.Session, orgID int64) (string, error) {
-	for i := 0; i < 3; i++ {
-		uid := util.GenerateShortUID()
-
-		exists, err := sess.Where("org_id=? AND uid=?", orgID, uid).Get(&accesscontrol.Role{})
-		if err != nil {
-			return "", err
-		}
-
-		if !exists {
-			return uid, nil
-		}
+func generateManagedRoleUID(orgID int64, name string) (string, error) {
+	parts := strings.Split(name, ":")
+	if len(parts) != 4 {
+		return "", fmt.Errorf("unexpected role name: %s", name)
 	}
-
-	return "", fmt.Errorf("failed to generate uid")
+	return fmt.Sprintf("managed_%d_%s_%s", orgID, parts[1], parts[2]), nil
```

Comment:
so instead of generating a GenerateShortUID() we now make a "uid" of the org_id and the parts.

This does not make a conflict either? 🤔 

thinking of adding some random string to the UID as well

Reply:
There is no need to add some random string to the uid. The org id will help to make them unique across orgs. So a managed role for e.g. a user 1 in org 1 would be `managed_1_users_1`

Reply:
i see. 👍 

could we add a test for generating this UID as an example OR document/comment an example of this UID. Cause I do not read directly that the second index `part[1]` is a userID.